src/docmanager/index.ts

```diff
@@ -437,7 +437,7 @@ class DocumentManager implements IDisposable {
       if (this._defaultWidgetFactory === name) {
         this._defaultWidgetFactory = '';
       }
-      for (let opt in Object.keys(this._defaultWidgetFactories)) {
+      for (let opt of Object.keys(this._defaultWidgetFactories)) {
```

Comment:
Why use `of Object.keys(this._defaultWidgetFactories))` instead of `in this._defaultWidgetFactories)` like you do elsewhere?